extension/config.go

```diff
@@ -0,0 +1,132 @@
+package extension
+
+import (
+	"os"
+	"strconv"
+	"time"
+)
+
+const (
+	// default buffering options for logs api
+	defaultTimeoutMS = 1000
+	defaultMaxBytes  = 262144
+	defaultMaxItems  = 1000
+
+	// Waiting too long to send a batch of events can be
+	// expensive in Lambda. It's reasonable to expect a
+	// batch send to complete in this amount of time.
+	defaultBatchSendTimeout = time.Second * 15
+
+	// It's very generous to expect an HTTP connection to
+	// to be established in this time.
+	defaultConnectTimeout = time.Second * 3
+)
+
+type Config struct {
+	ApiKey                         string // Honeycomb API key
```

Comment:
Nicer to use same name as old config struct - `APIKey`

Reply:
How much nicer is keeping the name if the previous Config struct never shipped? 

I think it boils down to which naming style would we like from here on out.

Reply:
For further comparison, this [replaces](https://github.com/honeycombio/honeycomb-lambda-extension/pull/112/files#diff-2873f79a86c0d8b3335cd7731b0ecf7dd4301eb19a82ef7a1cba7589b5252261L44-L46) an `apiKey` variable that used to be set in `main`.

Reply:
Well, [we're all over the place](https://github.com/search?l=Go&p=3&q=org%3Ahoneycombio+apikey&type=Code) as to a convention for "Api" vs "API". Let's go with your suggestion for "API" here. 👍🏻